### ISSUE TITLE
Fails to install with uv

### ISSUE THREAD

**Issue by jku** [sigstore/protobuf-specs#288]
https://github.com/astral-sh/uv currently fails to install protobuf-specs: https://github.com/theupdateframework/tuf-on-ci/issues/205
* uv does not accept pre-releases without an explicit `--prerelease=allow`
* this project depends on betterproto 2.0.0b6 

It's been about four years since 2.0.0b1 so waiting for betterproto to release might not be a good strategy. If we consider this important we could offer to do some work to the maintainer if that helps them make a release?

**Comment by woodruffw** [comment]
> It's been about four years since 2.0.0b1 so waiting for betterproto to release might not be a good strategy. If we consider this important we could offer to do some work to the maintainer if that helps them make a release?

Yeah, this would be ideal -- I've sent a few patches upstream to `betterproto` (in particular, fixing the handling of the `Struct` WKT), but I don't know how close they are to a stable 2.0 release.

(In particular, I still need to complete the fix in https://github.com/danielgtaylor/python-betterproto/pull/559, which I'm still slightly murky on.)

More generally though, prereleases are part of PEP 440 and `uv` *should* support them, at least when fully pinned (e.g. `protobuf-specs==2.0.0b6`). So I think it might *also* be good to prod `uv` into finishing their prerelease support 🙂

**Comment by spencerschrock** [comment]
This is also affecting users installing the [model-signing](https://pypi.org/project/model-signing/) library (from `sigstore/model-transparency`) either directly or indirectly with `uv`. 

e.g. https://github.com/Kaggle/kagglehub/issues/224#issuecomment-2657180882

As mentioned in the original comment, 2.0 of `betterproto` isn't known, and `uv` has stricter ideas of pre-release.
https://docs.astral.sh/uv/pip/compatibility/#pre-release-compatibility

> In the future, uv may support pre-release identifiers in transitive dependencies. However, it's likely contingent on evolution in the Python packaging specifications. The existing PEPs [do not cover "dependency resolution"](https://discuss.python.org/t/handling-of-pre-releases-when-backtracking/40505/17) and are instead focused on behavior for a single version specifier. As such, there are unresolved questions around the correct and intended behavior for pre-releases in the packaging ecosystem more broadly.

I don't follow PEPs too closely, so I don't know if anything has changed that would help `uv` "finish their prerelease support"?

**Comment by woodruffw** [comment]
I _think_ on the PEP side, the handling of precisely-pinned prereleases is well-defined -- `uv` _should_ be supporting `betterproto==2.0.0b6` (or whatever), since the user isn't asking for any prerelease resolution logic.

https://github.com/astral-sh/uv/issues/11483 seems to be tracking that as a more tractable variant of the prerelease resolution problem (which they're correct to note isn't well defined) 🙂

**Comment by stefanberger** [comment]
The author of betterproto suggested to migrate to betterproto2: https://github.com/danielgtaylor/python-betterproto/issues/678#issuecomment-3304419629

**Comment by woodruffw** [comment]
This is actually a non-issue for us now, since we've removed our dependency on protobufs/betterproto entirely in sigstore-python.

(This library itself is still not installable with uv, but IMO a better course of action here would be to deprecate the Python bindings here and point users towards [sigstore-models](https://github.com/astral-sh/sigstore-models/tree/main) as the alternative.)

**Comment by woodruffw** [comment]
Oops, didn't mean to close. This is still an issue in `sigstore-protobuf-specs`, just not in `sigstore-python`.
